projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -1206,6 +1353,26 @@ export class IgxComboComponent implements AfterViewInit, ControlValueAccessor, O
         }
     }
 
+    public ngAfterContentInit() {
+        if (this._itemTemplateProper) {
```

Comment:
So templates can only be assigned on initialization as further changes are ignored and must go though the API, more or less ignoring the dynamic updates from `@ContentChild`?

This sort of disables a simpler template configuration where you have one of the combo templates hidden with `ngIf` and a simple bool to toggle it, like the ones that used to be active in the dev demo:
```html
<ng-template *ngIf="currentDataType !== 'primitive'" #itemTemplate let-display let-key="valueKey">
    <!-- -->
</ng-template>
<ng-template *ngIf="currentDataType === 'primitive'" #itemTemplate let-display let-key="valueKey">
    <!-- -->
</ng-template>
```